php_phongo.c

```diff
@@ -2471,7 +2471,7 @@ static char* php_phongo_concat_handshake_data(const char* default_value, const c
 		ret_len += custom_value_len + PHONGO_METADATA_SEPARATOR_LEN;
 	}
 
-	ret = ecalloc(sizeof(char*), ret_len);
+	ret = ecalloc(ret_len, sizeof(char*));
```

Comment:
I think the second argument should be `sizeof(char)`, since we're allocating a string here and not an array of string pointers. In practice, this just means we've been allocating more memory than necessary (likely 8x as much on a 64-bit system). It looks like this mistake dates back to https://github.com/mongodb/mongo-php-driver/commit/895fcf4bbc0f70eb7537fb8ac62610840849af7b#diff-14fe3819169577796c8c63f3bea25bcf0504df8c6e131f7beb916d255c3d49d8R2499.



@alcaeus can you confirm?

Reply:
You are correct (in both cases). This should indeed by `sizeof(char)`.
```suggestion
	ret = ecalloc(ret_len, sizeof(char));
```